File: ScheduleFaculty.Core/Entities/HourStudyOfAYear.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ScheduleFaculty.Core.Entities;

public class HourStudyOfAYear
{
    public Guid Id { get; set; }

    [ForeignKey(("CourseHourType"))] 
    public Guid CourseHourTypeId { get; set; }
    public CourseHourType CourseHourType { get; set; }
    
    [ForeignKey(("User"))] 
    public string UserId { get; set; }
    public ApplicationUser User { get; set; }
    
    [ForeignKey(("Classroom"))] 
    public Guid ClassroomId { get; set; }
    public Classroom Classroom { get; set; }
    
    public List<int> StudyWeeks { get; set; }
    
    public DayOfWeek DayOfWeek { get; set; }
    
    public int StartTime { get; set; }
    
    public int EndTime { get; set; }
    
    
}

File: Frontend/src/app/models/hour-study-of-ayear.ts
import {WeekDay} from "@angular/common";
import {IClassroom} from "./classroom";
import {IUser} from "./login";
import {ICourseHourTypeResponse} from "./course-hour-type";
import {IStudyYearGroup} from "./study-year-group";

export interface IHourStudyOfAYear {
  id: string | null,
  semiGroupsId: string[] | null,
  courseHourTypeId: string | null,
  userId: string | null,
  classroomId: string | null,
  studyWeeks: number[] | null,
  dayOfWeek: WeekDay | null,
  startTime: number | null,
  endTime: number | null,
}

export interface IHourStudyOfAYearResponse {
  id: string,
  semiGroups: IStudyYearGroup[],
  courseHourType: ICourseHourTypeResponse,
  user: IUser,
  classroom: IClassroom,
  studyWeeks: number[],
  dayOfWeek: WeekDay,
  startTime: number,
  endTime: number,
}
